contracts/VToken.sol

```diff
@@ -1509,6 +1576,12 @@ contract VToken is
         emit NewProtocolShareReserve(oldProtocolShareReserve, address(protocolShareReserve_));
     }
 
+    function _ensureSenderIsDelegateOf(address user) internal view {
+        if (!ComptrollerViewInterface(address(comptroller)).approvedDelegates(user, msg.sender)) {
```

Comment:
Should we always assume that the user is the delegate of themselves? Like

```suggestion
        if (user != msg.sender && !ComptrollerViewInterface(address(comptroller)).approvedDelegates(user, msg.sender)) {
```

Not sure about that tbh, may or may not be useful for integrations

Reply:
I believe we should avoid incorporating it if a user intends to utilize `borrowOnBehalf` or similar `onBehalf` functionalities for their own. Instead, they ought to use the standard borrow or other corresponding functions. 
Should a user wish to proceed, they must initially grant themselves approval before being permitted to do so. Likewise, we have not made this assumption within the core pool.